pkg/kubernetes/podexec/v1alpha1/podexec.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/ghodss/yaml"
+	"github.com/openebs/maya/pkg/template"
+	"github.com/pkg/errors"
+	api_core_v1 "k8s.io/api/core/v1"
+)
+
+type podexec struct {
+	object       *api_core_v1.PodExecOptions
+	ignoreErrors bool
+	errs         []error
+}
+
+// AsAPIPodExec validate and returns PodExecOptions object pointer and error
+// depending on ignoreErrors opt and errors
+func (p *podexec) AsAPIPodExec() (podExecOptions *api_core_v1.PodExecOptions, err error) {
+	err = p.Validate()
+	if err != nil && !p.ignoreErrors {
+		return nil, err
+	}
+	return p.object, nil
+}
+
+// WithTemplate takes Yaml values which is given in runtask and key in which configuration
+// is present and unmarshal it with PodExecOptions.
+func WithTemplate(context, yamlString string, values map[string]interface{}) (p *podexec, err error) {
```

Comment:
i felt avoiding error from the signature would have made the calling code readable.
I felt using err as a value will be good.
in other words if err was captured into `errs` property; then calling code would look like:
```go
e, err := podexec.WithTemplate(abc, yml, nil).AsAPIPodExec()
// or 
e, err := podexec.WithTemplate(xyz, yaml, nil).Apply(myopts).AsAPIPodExec()
```

Above will not force calling code to keep checking for err everytime a method is invoked.
Assumption is some of the methods are used to build the object while some of the methods are invoked finally.
Here `AsAPIPodExec` is the final method.

However, this comment about error handling is optional & should be considered on a case-by-case basis.

Reply:
done